cmd/soroban-rpc/internal/preflight/pool.go

```diff
@@ -26,18 +27,17 @@ type PreflightWorkerPool struct {
 	ledgerEntryReader db.LedgerEntryReader
 	networkPassphrase string
 	logger            *log.Entry
-	done              chan struct{}
+	isClosed          atomic.Bool
 	requestChan       chan workerRequest
 	wg                sync.WaitGroup
 }
 
-func NewPreflightWorkerPool(workerCount uint, ledgerEntryReader db.LedgerEntryReader, networkPassphrase string, logger *log.Entry) *PreflightWorkerPool {
+func NewPreflightWorkerPool(workerCount uint, jobQueueCapacity uint, ledgerEntryReader db.LedgerEntryReader, networkPassphrase string, logger *log.Entry) *PreflightWorkerPool {
```

Comment:
There's probably a refactor we could do (at some point), to:
```go
func NewJsonRPCMethodWorkerPool(
  workerCount uint,
  jobQueueuCapacity uint,
  handler jrpc2.Handler,
) *WorkerPool
```
Or even using generics for the handler 🙈

Reply:
I had actually started some work work in those lines at https://github.com/2opremio/soroban-cli/commit/afa1c5daff32c19516026071b363f6b5fa6945e2 .